packages/apps/types/src/schema/collection.ts

```diff
@@ -0,0 +1,15 @@
+//
+// Copyright 2024 DXOS.org
+//
+
+import { Expando, ref, S, TypedObject } from '@dxos/echo-schema';
+
+export class Collection extends TypedObject({ typename: 'dxos.Collection', version: '0.1.0' })({
+  name: S.optional(S.string),
+  objects: S.mutable(S.array(ref(Expando))),
+  // Key is schema typename and value is reference to a view object of the associated schema.
+  // Having collection reference the views rather than vice versa ensures that the state converges to a single view per key (i.e. type).
+  // This also leaves open a future where this key could be changed to allow for multiple stack views per section.
+  // TODO(wittjosiah): Any way to make this more type safe?
+  views: S.mutable(S.record(S.string, ref(Expando))),
```

Comment:
What kinds of objects can `views` reference?

Reply:
At first glance, this seems like it should be much less permissive than `objects`.

Reply:
The object for the view should be of the schema associated with the `typename` of the record key. Would be nice to find a way to have that be more strongly typed but I didn't want to block on that. Open to ideas here though